superset/assets/src/SqlLab/components/LimitControl.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button,
+  Label,
+  FormGroup,
+  FormControl,
+  Overlay,
+  Popover,
+} from 'react-bootstrap';
+import ControlHeader from '../../explore/components/ControlHeader';
+import { t } from '../../locales';
+
+const propTypes = {
+  value: PropTypes.number,
+  defaultQueryLimit: PropTypes.number.isRequired,
+  maxRow: PropTypes.number.isRequired,
+  onChange: PropTypes.func.isRequired,
+};
+
+export default class LimitControl extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    const { value, defaultQueryLimit } = props;
+    this.state = {
+      textValue: value.toString() || defaultQueryLimit.toString(),
+      showOverlay: false,
+    };
+  }
+
+  setValueAndClose(val) {
+    this.setState({ textValue: val }, this.submitAndClose.bind(this));
+  }
+
+  submitAndClose() {
+    const value = parseInt(this.state.textValue, 10) || this.props.defaultQueryLimit;
+    this.props.onChange(value);
+    this.setState({ showOverlay: false });
+  }
+
+  isValidLimit(limit) {
+    const value =  parseInt(limit, 10);
+    return !(isNaN(value) || value <= 0 || (this.props.maxRow && value > this.props.maxRow));
+  }
+
+  handleToggle() {
+    this.setState({ showOverlay: !this.state.showOverlay });
+  }
+
+  handleHide() {
+    this.setState({ showOverlay: false });
+  }
+
+  renderPopover() {
+    const textValue = this.state.textValue;
+    const isValid = this.isValidLimit(textValue);
+    const errorMsg = 'Row limit must be positive integer' +
+      (this.props.maxRow ? ` and below ${this.props.maxRow}` : '');
+    return (
+      <Popover id="sqllab-limit-results">
```

Comment:
Is the `id` necessary?

Reply:
This is primarily for getting rid of the warning. Do we want to remove it?